healthcheck.go

```diff
@@ -1,118 +1,78 @@
 package main
 
 import (
-	"encoding/json"
-	"errors"
-	"fmt"
-	"io/ioutil"
 	"net/http"
 
-	"github.com/Financial-Times/go-fthealth"
+	fthealth "github.com/Financial-Times/go-fthealth/v1_1"
 	"github.com/Financial-Times/message-queue-go-producer/producer"
 	"github.com/Financial-Times/message-queue-gonsumer/consumer"
+	"github.com/Financial-Times/service-status-go/gtg"
 )
 
-// Healthcheck offers methods to measure application health.
-type Healthcheck struct {
-	client   http.Client
-	srcConf  consumer.QueueConfig
-	destConf producer.MessageProducerConfig
+type HealthCheck struct {
+	consumer consumer.MessageConsumer
+	producer producer.MessageProducer
 }
 
-func (h *Healthcheck) checkHealth() func(w http.ResponseWriter, r *http.Request) {
-	return fthealth.HandlerParallel("Dependent services healthcheck", "Checks if all the dependent services are reachable and healthy.", h.messageQueueProxyReachable())
+func NewHealthCheck(p producer.MessageProducer, c consumer.MessageConsumer) *HealthCheck {
+	return &HealthCheck{
+		consumer: c,
+		producer: p,
+	}
 }
 
-func (h *Healthcheck) gtg(writer http.ResponseWriter, req *http.Request) {
-	healthChecks := []func() error{h.checkAggregateMessageQueueProxiesReachable}
-
-	for _, hCheck := range healthChecks {
-		if err := hCheck(); err != nil {
-			writer.WriteHeader(http.StatusServiceUnavailable)
-			return
-		}
+func (h *HealthCheck) Health() func(w http.ResponseWriter, r *http.Request) {
+	checks := []fthealth.Check{h.readQueueCheck(), h.writeQueueCheck()}
+	hc := fthealth.HealthCheck{
+		SystemCode:  "v1-suggestor",
+		Name:        "V1 Suggestor",
+		Description: "Checks if all the dependent services are reachable and healthy.",
+		Checks:      checks,
 	}
+	return fthealth.Handler(hc)
 }
 
-func (h *Healthcheck) messageQueueProxyReachable() fthealth.Check {
+func (h *HealthCheck) readQueueCheck() fthealth.Check {
 	return fthealth.Check{
-		BusinessImpact:   "Content V1 Metadata is not suggested. This will negatively impact V1 metadata availability.",
-		Name:             "MessageQueueProxyReachable",
-		PanicGuide:       "https://sites.google.com/a/ft.com/technology/systems/dynamic-semantic-publishing/extra-publishing/v1-suggestor-runbook",
+		ID:               "read-message-queue-proxy-reachable",
+		Name:             "Read Message Queue Proxy Reachable",
 		Severity:         1,
-		TechnicalSummary: "Message queue proxy is not reachable/healthy",
-		Checker:          h.checkAggregateMessageQueueProxiesReachable,
+		BusinessImpact:   "Content V1 Metadata is not suggested. This will negatively impact V1 metadata availability.",
```

Comment:
Business Impact message is not clear enough. Suggestion: "Content V1 Metadata can't be read from queue. This will negatively impact V1 metadata availability."